src/Tools/RepoUtil/GenerateUtil.cs

```diff
@@ -24,11 +24,18 @@ internal static class GenerateUtil
         /// </summary>
         internal static List<NuGetPackage> GetFilteredPackages(GenerateData generateData, RepoData repoData)
         {
-            // Fixed packages are never included in generated output.  Doing so would create conflicts because it's
-            // possible for two versions to exist for the same package.  Take for example System.Collections.Immuatble
-            // which is both fixed and floating in Roslyn.
+            // Only fixed packages with 'generate names' are included in generated output. Doing
+            // otherwise would create conflicts because it's possible for two versions to exist
+            // for the same package.  Take for example System.Collections.Immuatble which is both
+            // fixed and floating in Roslyn.
+            var fixedWithGenerate = repoData
+                .FixedPackages
+                .Where(x => x.GenerateNameOpt != null)
+                .Select(x => new NuGetPackage(x.GenerateNameOpt, x.Package.Version));
```

Comment:
I'm nervous about this approach.  It's creating a NuGetPackage object which not correct: a fake name paired with a version.  Consider a case where the fake name actually points to a real package.  Could cause issues down the line. 

I think we should extend NuGetpPackage to have the concept of GenerateName.  Defaults to package name but can be overriden.